src/ray/raylet/worker_pool.h

```diff
@@ -221,6 +221,9 @@ class WorkerPool {
   /// for a given language.
   State &GetStateForLanguage(const Language &language);
 
+  /// Add a time for the starting worker process to avoid zombie worker process.
+  void AddTimerForStartingWorkerProcess(const Process &proc, const Language &language);
```

Comment:
```suggestion
  /// Start a timer to monitor the starting worker process. 
  ///
  /// If any workers in this process don't register within the timeout (due to worker process crash or any other reasons), 
  /// remove them `starting_worker_processes`. Otherwise if we'll mistakenly think there are unregistered workers, and won't start new workers.
  void MonitorStartingWorkerProcess(const Process &proc, const Language &language);
```
What about this?